rest/config_test.go

```diff
@@ -107,3 +108,42 @@ func TestLoadServerConfigExamples(t *testing.T) {
 	})
 	assert.NoError(t, err)
 }
+
+// Test TLS Version
+func TestTLSVersionSetting(t *testing.T) {
+	tests := []struct {
+		name        string
+		config      string
+		expectedTLS uint16
+	}{
+		{
+			name:        `Set TLS 1.0`,
+			config:      `{"tls_minimum_version": "tlsv1", "SSLCert": "examples/ssl/cert.pem", "SSLKey": "examples/ssl/privkey.pem"}`,
+			expectedTLS: tls.VersionTLS10,
+		},
+		{
+			name:        `Set TLS 1.1`,
+			config:      `{"tls_minimum_version": "tlsv1.1", "SSLCert": "examples/ssl/cert.pem", "SSLKey": "examples/ssl/privkey.pem"}`,
+			expectedTLS: tls.VersionTLS11,
+		},
+		{
+			name:        `Set TLS 1.2`,
+			config:      `{"tls_minimum_version": "tlsv1.2", "SSLCert": "examples/ssl/cert.pem", "SSLKey": "examples/ssl/privkey.pem"}`,
+			expectedTLS: tls.VersionTLS12,
+		},
+		{
+			name:        `No TLS set, should default to 1.0`,
+			config:      `{"SSLCert": "examples/ssl/cert.pem", "SSLKey": "examples/ssl/privkey.pem"}`,
+			expectedTLS: tls.VersionTLS10,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			buf := bytes.NewBufferString(test.config)
+			config, _ := readServerConfig(SyncGatewayRunModeNormal, buf)
```

Comment:
Could probably get away with avoiding the config loading for this test, and have a pure unit test that just compares inputs/outputs for `GetTLSVersionFromString`.